src/Workspaces/MSBuildTest/VisualStudioMSBuildInstalled.cs

```diff
@@ -151,4 +98,106 @@ private static void RegisterMSBuildAssemblyResolution(string msbuildToolsPath)
             };
         }
     }
+
+    internal class VisualStudio16_2OrHigherMSBuildInstalled : VisualStudioMSBuildInstalled
+    {
+        public VisualStudio16_2OrHigherMSBuildInstalled() : base(new Version(16, 2))
+        {
+        }
+    }
+
+    internal static class VisualStudioMSBuildLocator
```

Comment:
I pulled this out to isolate the "have we already checked" cache since it was getting a bit screwy with all of the static bits. Unfortunately this made the diff icky.